From e1c1fc73e222ba92e9f9b6d6d75109ca793be0ea Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Thu, 26 Apr 2012 22:50:11 +0200
Subject: [PATCH] windowsNavigator: only access tooltip if it's known to exist

Apparently, windowOverlay._text can be null sometimes. In that
case, do nothing instead of crashing.
---
 extensions/windowsNavigator/extension.js |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/extensions/windowsNavigator/extension.js b/extensions/windowsNavigator/extension.js
index 3ec4abf..35188e3 100644
--- a/extensions/windowsNavigator/extension.js
+++ b/extensions/windowsNavigator/extension.js
@@ -34,7 +34,7 @@ function enable() {
     resetState();
 
     Workspace.WindowOverlay.prototype.setId = function(id) {
-        if (this._text.visible && id == null)
+        if (this._text && this._text.visible && id == null)
             this._text.hide();
         this._id = id;
         if (id != null)
@@ -57,7 +57,7 @@ function enable() {
     winInjections['showTooltip'] = undefined;
 
     Workspace.WindowOverlay.prototype.hideTooltip = function() {
-        if (this._text.visible)
+        if (this._text && this._text.visible)
             this._text.hide();
     }
     winInjections['hideTooltip'] = undefined;
-- 
1.7.8.6

